### DISCUSSION TITLE
ZipChord 2.1 and beyond

### DISCUSSION THREAD

**Discussion by psoukie** [psoukie/zipchord#145] [Announcements]
The following weeks will be a bit slow because of a combination of a workload in my day job and a medical emergency in my family. But to share where we are and where we are going next:

I have sorted most of the [existing issues](https://github.com/psoukie/zipchord/issues) into "Core Improvements" and "Non-Dependent Improvements". If your feature request or issue is assigned to the "Core Improvement" milestone, it means it is dependent on two things: <img align="right" src="https://user-images.githubusercontent.com/42703196/212493161-375d7c55-1b26-40a5-8daf-9188d8702010.png" width=400>

1. **Test automation** that would let me test that whatever changes are being made, they do not cause issues in various configurations or combinations of settings. The good news is the automation was done, and is part of the console in the "Developer" version of ZipChord 2.1 Beta. (Even though probably of limited use to regular users.)
2. **Refactoring of the code** in line with this diagram. Sadly, the current workings of large portions of the code (especially the additions of shorthands and hints) were pasted on top of existing code rather hastily. The refactoring itself was dependent on the test automation, but it's the thing I'll be working on next.

The reasons I prioritized the testing automation and redesigning the code is to avoid putting a quick and dirty solution on top of another quick and dirty solution, and to have a mechanism to test that my changes are not breaking something else.

The "Non-Dependent Improvements" -- as the name suggests -- are those that can be added without risking introducing errors into the core workings of ZipChord, and I can include them in parallel with some of the core changes. An example of these are the application keyboard shortcuts and context-sensitive help in ZipChord 2.1.

These changes will allow easier integration of more possibilities without increasing complexity for the users. (Some of the things discussed under the topic about context chording in #101 show both simple and more advanced things that could be done.)

In the meantime, feel free to grab the [2.1 Beta](https://github.com/psoukie/zipchord/releases/tag/2.1.0-beta) and let me know what you think. (The source code under the main branch already includes a few small fixes and improvements on top of the Beta.)

**Comment by Nooby-001** [comment]
Wishing the best for you at home and work. 😀
